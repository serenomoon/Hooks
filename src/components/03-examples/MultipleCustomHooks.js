import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch'

export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);

    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    const {author, quote} = !!data && data[0];
//    console.log(state);
  return (
    
    <>
        <h1> Breaking Bad Quotes </h1>
        <hr />

        {
            loading ?
            (<div className='alert alert-info text-center'>
                Loading...
            </div>)
            :
            (<><blockquote className='blockquote text-right'>
                <p className='mb-0'>{quote}</p>
                <br></br>
                <footer className='blockquote-footer'>{author}</footer>
            </blockquote>

            <button 
            className='btn btn-primary'
            onClick={increment}
            >Next Quote</button>
            </>)
        }
    </>

  )
}
