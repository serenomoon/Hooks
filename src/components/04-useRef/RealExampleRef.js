import React, { useState } from 'react'
import { MultipleCustomHooks }from '../03-examples/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(false)

  return (
    <div>
        <h1>RealExampleRef</h1>
        <hr />

        { show && <MultipleCustomHooks />}

        <button
            className='btn btn-primary'
            onClick={ () => { setShow(!show) } }
        >Show/Hide</button>
    </div>
  )
}
