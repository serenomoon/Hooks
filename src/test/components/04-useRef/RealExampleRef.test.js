import React, { useState } from 'react';
import { shallow } from 'enzyme';
import { renderHook, act } from "@testing-library/react-hooks";
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';



describe('Test "RealExamplesRef.js" ', () => {
    
    test('should show correct snapshot', () => {
        
        const wrapper = shallow( <RealExampleRef /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    });

    test('should show "<MultipleCustomHooks />" component', () => {
        

        const wrapper = shallow( <RealExampleRef /> );

        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    });

});