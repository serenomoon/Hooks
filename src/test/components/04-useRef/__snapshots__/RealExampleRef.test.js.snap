// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test "RealExamplesRef.js"  should show correct snapshot 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
